lib/blueprinter/base.rb

```diff
@@ -124,14 +124,21 @@ def self.field(method, options = {}, &block)
     #   end
     #
     # @return [Field] A Field object
-    def self.association(method, options = {})
+    def self.association(method, options = {}, &block)
       raise BlueprinterError, 'blueprint required' unless options[:blueprint]
       name = options.delete(:name) || method
+
+      options = if block_given?
+        options.merge(extractor: BlockExtractor, block: block)
+      else
+        options.merge(extractor: AutoExtractor)
+      end
+
       current_view << Field.new(method,
-                                       name,
-                                       AssociationExtractor,
-                                       self,
-                                       options.merge(association: true))
+                                name,
+                                AssociationExtractor.new(options[:extractor]),
```

Comment:
I don't think you need to pass the `options[:extractor]` here, the `AssociationExtractor` should already be passed the `options` from `Field` https://github.com/procore/blueprinter/blame/master/lib/blueprinter/field.rb#L13